src/main/java/org/glenrockindiancommunity/controller/HomeController.java

```diff
@@ -18,4 +27,36 @@ public String redirectToIndex(Model model) {
     return "redirect:/index.jsp";
   }
 
+  @RequestMapping(value = "/login")
+  public String loginPage() {
+    return "redirect:/login.jsp";
+  }
+
+  @RequestMapping(value = "/logout")
+  public String logoutPage(HttpServletRequest request, HttpServletResponse response) {
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    if (auth != null) {
+      new SecurityContextLogoutHandler().logout(request, response, auth);
+    }
+    return "redirect:/login?logout";//You can redirect wherever you want, but generally it's a good idea to show login screen again.
+  }
+
+  @RequestMapping(value = "/admin", method = RequestMethod.GET)
+  public String adminPage(ModelMap model) {
+    model.addAttribute("user", getPrincipal());
+    return "redirect:/admin.jsp";
+  }
+
+  private String getPrincipal() {
+    String userName = null;
+    Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+    if (principal instanceof UserDetails) {
+      userName = ((UserDetails) principal).getUsername();
+    } else {
+      userName = principal.toString();
+    }
+    return userName;
+  }
+
 }
```

Comment:
With spring-boot, you don't need to do so much config. This is a mix of old and new style.